app/src/main/java/de/westnordost/streetcomplete/screens/settings/SettingsFragment.kt

```diff
@@ -65,6 +67,8 @@ class SettingsFragment :
     private val questTypeRegistry: QuestTypeRegistry by inject()
     private val visibleQuestTypeSource: VisibleQuestTypeSource by inject()
     private val questPresetsSource: QuestPresetsSource by inject()
+    private val osmQuestsHiddenDao: OsmQuestsHiddenDao by inject()
+    private val noteQuestsHiddenDao: NoteQuestsHiddenDao by inject()
```

Comment:
It is uncommon (if not: not done at all) in this codebase to directly access the Daos from view code. Fragments and Activities instead access *Controllers or *Source classes to update or get data. The update of the view subsequently should be reactive to changes in the data, i.e. the quest summary should not update after having tapped the button, but instead the tapping of the button changes the data and because the data is changed, the label is updated.

I wanted to write the details but that got so involved that I rather do it myself unless you know now exactly what to do. I'll do that then otherwise shortly before the next update.

Reply:
I'd be happy if you could work on this as I don't think I'm expert enough to do it myself. Thank you!